### ISSUE TITLE
Hexagonal Grid System

### ISSUE THREAD

**Issue by ZuhairGhias** [AMastryukov/LD49#2]
We need a hexagonal grid system that allows us to grab, drag and place hexagonal grid tiles around on the map.
This grid system should have a way to track all tiles.

I think it would be best to track all tiles by registering and de-registering them using a `OnHexPlaced` and `OnHexRemoved` events. A `HexMapManager` can handle the tracking of tiles and registering these events.

**Comment by AMastryukov** [comment]
```
HexMapManager
- HexTile OriginTile => return Tiles[0];
- List<HexTile> Tiles

HexTile:
- TileData (ScriptableObject)
- List<HexTile> NeighborTiles
```

**Comment by AMastryukov** [comment]
**Requirements:**
- [ ] Each HexTile should know its neighbors (6)
- [ ] Each HexTile should contain a TileData (ScriptableObject)
- [ ] Each HexTile should contain the model of the hex tile (the visual building)
- [ ] Each HexTile should only be placed 
- [ ] HexMapManager should always track all HexTiles on the map using events

**Comment by ZuhairGhias** [comment]
Fixed in #18
